bin/bout-v5-factory-upgrader.py

```diff
@@ -170,6 +171,21 @@ def fix_deletions(variables, source):
     return source
 
 
+def fix_create_method(factory, source):
+    """Fix change of name of factory `create` method
+
+    """
+
+    if "old_create_method" not in factory:
+        return source
+    return re.sub(
+        r"({factory_name})\s*::\s*{old_create_method}\b".format(**factory),
+        r"\1::{create_method}".format(**factory),
```

Comment:
Can we check that there's no argument to the `Create` method? The old version could take a `Mesh*` argument, which I'd be surprised if anyone was using, but if they were it'd be an error to pass as the first argument to the new `create` (which is an `Options*`). Don't think it's worth bothering to fix if there was an argument, but raising an error or a warning might be nice?

Reply:
Done. It's a bit dumb, in that it always warns if a factory where the arguments have changed is used. Hopefully the documentation is sufficiently useful for users to know what to do, otherwise we'll have to rely on the compiler. For me, using gcc 10.2, I get the following error (with nice colours too):

```
../tests/integrated/test-invpar/test_invpar.cxx: In function ‘int main(int, char**)’:
../tests/integrated/test-invpar/test_invpar.cxx:33:32: error: cannot convert ‘Mesh*’ to ‘Options*’
   33 |   auto inv = InvertPar::create(mesh);
      |                                ^~~~
      |                                |
      |                                Mesh*
In file included from ../tests/integrated/test-invpar/test_invpar.cxx:8:
../include/invert_parderiv.hxx:104:53: note:   initializing argument 1 of ‘static std::unique_ptr<InvertPar> InvertPar::create(Options*, CELL_LOC, Mesh*)’
  104 |   static std::unique_ptr<InvertPar> create(Options *opt_in = nullptr,
      |                                            ~~~~~~~~~^~~~~~~~~~~~~~~~
```

Reply:
looks good to me, thanks @ZedThree!